### ISSUE TITLE
BUG `drizzle-kit introspect:mysql` from Planetscale db returns wrong primary key

### ISSUE THREAD

**Issue by pbochis** [drizzle-team/drizzle-kit-mirror#216]
Running `drizzle-kit introspect:mysql` on a planetscale database returns wrong primary key. 
This bug also breaks `drizzle-kit push:mysql` because the schema diff generated before trying to make changes is wrong.

I've tried to drop the entire planetscale database, recreate it from `schema.ts` with `drizzle-kit push:mysql`, and then do another introspect right after to check the result. It came out with wrong primary keys again.

Doing the above process with a local docker mysql worked fine, with no issue in pulling the primary keys.

Actual schema, generated with running `show create table [tableName]`:
```
CREATE TABLE `People`
(
    `PersonId`   int NOT NULL AUTO_INCREMENT,
    `FirstName`  varchar(50) DEFAULT NULL,
    `LastName`   varchar(50) DEFAULT NULL,
    `Employment` varchar(50) DEFAULT NULL,
    `Email`      varchar(50) DEFAULT NULL,
    `ClerkId`    varchar(40) DEFAULT NULL,
    PRIMARY KEY (`PersonId`),
    UNIQUE KEY `email_uniq_idx` (`Email`),
    KEY          `clerk_idx` (`ClerkId`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci

CREATE TABLE `TrainingRegistrations`
(
    `TrainingRegistrationId` int  NOT NULL AUTO_INCREMENT,
    `TrainingId`             int  NOT NULL,
    `PersonId`               int  NOT NULL,
    `RegistrationDate`       date NOT NULL,
    PRIMARY KEY (`TrainingRegistrationId`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci

CREATE TABLE `Trainings`
(
    `TrainingId`       int  NOT NULL AUTO_INCREMENT,
    `StartDate`        date NOT NULL,
    `EndDate`          date NOT NULL,
    `City`             varchar(50) DEFAULT NULL,
    `Venue`            varchar(50) DEFAULT NULL,
    `TrainingType`     varchar(50) DEFAULT NULL,
    `TrainingCategory` varchar(50) DEFAULT NULL,
    PRIMARY KEY (`TrainingId`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
```

Schema generated by `drizzle-kit introspect:mysql`

```
CREATE TABLE `People` (
	`PersonId` int AUTO_INCREMENT NOT NULL,
	`FirstName` varchar(50),
	`LastName` varchar(50),
	`Employment` varchar(50),
	`Email` varchar(50),
	`ClerkId` varchar(40),
	CONSTRAINT `People_PersonId` PRIMARY KEY(`PersonId`),
	CONSTRAINT `email_uniq_idx` UNIQUE(`Email`)
);
--> statement-breakpoint
CREATE TABLE `TrainingRegistrations` (
	`TrainingRegistrationId` int AUTO_INCREMENT NOT NULL,
	`TrainingId` int NOT NULL,
	`PersonId` int NOT NULL,
	`RegistrationDate` date NOT NULL,
	CONSTRAINT `TrainingRegistrations_PersonId` PRIMARY KEY(`PersonId`)
);
--> statement-breakpoint
CREATE TABLE `Trainings` (
	`TrainingId` int AUTO_INCREMENT NOT NULL,
	`StartDate` date NOT NULL,
	`EndDate` date NOT NULL,
	`City` varchar(50),
	`Venue` varchar(50),
	`TrainingType` varchar(50),
	`TrainingCategory` varchar(50),
	CONSTRAINT `Trainings_PersonId` PRIMARY KEY(`PersonId`)
);
--> statement-breakpoint
CREATE INDEX `clerk_idx` ON `People` (`ClerkId`);
```

And the typescript schema, generated by drizzle

```
export const people = mysqlTable("People", {
	personId: int("PersonId").autoincrement().notNull(),
	firstName: varchar("FirstName", { length: 50 }),
	lastName: varchar("LastName", { length: 50 }),
	employment: varchar("Employment", { length: 50 }),
	email: varchar("Email", { length: 50 }),
	clerkId: varchar("ClerkId", { length: 40 }),
},
(table) => {
	return {
		clerkIdx: index("clerk_idx").on(table.clerkId),
		peoplePersonId: primaryKey(table.personId),
		emailUniqIdx: unique("email_uniq_idx").on(table.email),
	}
});

export const trainingRegistrations = mysqlTable("TrainingRegistrations", {
	trainingRegistrationId: int("TrainingRegistrationId").autoincrement().notNull(),
	trainingId: int("TrainingId").notNull(),
	personId: int("PersonId").notNull(),
	// you can use { mode: 'date' }, if you want to have Date as type for this column
	registrationDate: date("RegistrationDate", { mode: 'string' }).notNull(),
},
(table) => {
	return {
		trainingRegistrationsPersonId: primaryKey(table.personId),
	}
});

export const trainings = mysqlTable("Trainings", {
	trainingId: int("TrainingId").autoincrement().notNull(),
	// you can use { mode: 'date' }, if you want to have Date as type for this column
	startDate: date("StartDate", { mode: 'string' }).notNull(),
	// you can use { mode: 'date' }, if you want to have Date as type for this column
	endDate: date("EndDate", { mode: 'string' }).notNull(),
	city: varchar("City", { length: 50 }),
	venue: varchar("Venue", { length: 50 }),
	trainingType: varchar("TrainingType", { length: 50 }),
	trainingCategory: varchar("TrainingCategory", { length: 50 }),
},
(table) => {
	return {
		trainingsPersonId: primaryKey(table.personId),
	}
});
```

As you can see, `PersonId` is taken as primary key for all 3 tables, even if clearly it is not, and one of them doesn't even have such a column

**Comment by bdonchenko** [comment]
facing the same issue.

also when a UNIQUE constraint contains more than 1 field, introspect takes only 1 field to the schema:

<img width="486" alt="image" src="https://github.com/drizzle-team/drizzle-kit-mirror/assets/15363698/48c88c76-8048-41ec-9698-d15b8d13fb8a">

<img width="586" alt="image" src="https://github.com/drizzle-team/drizzle-kit-mirror/assets/15363698/e34e0fc4-e3a1-4c58-a44c-1029a2426c80">

**Comment by nic-southern** [comment]
Having this issue too, I posted (probably incorrectly) in drizzle-orm but here's my issue there:
https://github.com/drizzle-team/drizzle-orm/issues/1428 

Difference is seems like it doesn't even try to create the primary key on my "id" column.

**Comment by alexevladgabriel** [comment]
I guess we are all using PlanetScale, Next-Auth, because if we remove the base schema of auth, it would work like charm.

**Comment by theonlyway** [comment]
Just started getting this myself on PlanetScale. I'm not using Next-Auth though. I fully deleted my my development branch in Planetscale and re-created the tables and it worked. Run it again literally right after and it wants to drop all primary keys then add them back

**Comment by AndriiSherman** [comment]
@pbochis just tried to do the same with `drizzle-kit@0.20.6` and `drizzle-orm@0.29.0` and can't reproduce it with PlanetScale. I used your schema definitions and introspect worked well for me

**Comment by AndriiSherman** [comment]
If you will face same issues with latest versions - please ping me. I'll try to check it more

**Comment by MendyLanda** [comment]
> If you will face same issues with latest versions - please ping me. I'll try to check it more

@AndriiSherman I can confirm this issue exists on the latests version still (mysql, planetscale.)
initial push works fine and any subsequent push fails with:
```bash
Error: target: main.-.primary: vttablet: rpc error: code = InvalidArgument desc = Incorrect table definition; there can be only one auto column and it must be defined as a key (errno 1075) (sqlstate 42000) (CallerID: 9nm8fyuwnk9vhh4l1tqa): Sql: "alter table email_verification drop primary key", BindVars: {REDACTED}
    at PromiseConnection.query (/Users/mendy/dev/@kangaroo/node_modules/drizzle-kit/bin.cjs:44259:26)
    at Command.<anonymous> (/Users/mendy/dev/@kangaroo/node_modules/drizzle-kit/bin.cjs:63071:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ER_WRONG_AUTO_KEY',
  errno: 1075,
  sql: 'ALTER TABLE `email_verification` DROP PRIMARY KEY;',
  sqlState: '42000',
  sqlMessage: 'target: main.-.primary: vttablet: rpc error: code = InvalidArgument desc = Incorrect table definition; there can be only one auto column and it must be defined as a key (errno 1075) (sqlstate 42000) (CallerID: 9nm8fyuwnk9vhh4l1tqa): Sql: "alter table email_verification drop primary key", BindVars: {REDACTED}'
}
```
<img width="212" alt="image" src="https://github.com/drizzle-team/drizzle-kit-mirror/assets/54242706/e0aa0dc9-6986-41b7-805a-7503a36f7bd3">

**Comment by AndriiSherman** [comment]
@MendyLanda 
Could you please DM me in Discord or X or any other place? Would love to get your schema example and add some of that to test cases

After investigation, I'll respond here with results

**Comment by AndriiSherman** [comment]
@MendyLanda can reproduce now, thanks!

**Comment by AndriiSherman** [comment]
Fixed locally. Will be included in the next release

**Comment by iLegacy** [comment]
> Fixed locally. Will be included in the next release

Tysm 🙏

**Comment by MendyLanda** [comment]
> Fixed locally. Will be included in the next release

@AndriiSherman TYSM! 🙏.
Feel free to use the schema for testing if you so wish.

**Comment by dane-stevens** [comment]
@AndriiSherman Thank you! This is great!

**Comment by AndriiSherman** [comment]
So it should be fixed on `drizzle-kit@beta` now.

@dane-stevens for your specific case, you will still see a few migrations generated by drizzle-kit on every push. This one is related to your use of 

```typescript
.default(sql`CURRENT_TIMESTAMP(3) ON UPDATE CURRENT_TIMESTAMP(3)`)
```

I would not recommend using `ON UPDATE` in default unless you really need it and are fine with having those changes triggered on every push. I'll review this [PR](https://github.com/drizzle-team/drizzle-orm/pull/1114), add a custom onUpdate SQL, and make the necessary changes to drizzle-kit this week. This way, you can have native support for the custom onUpdate function.

**Comment by AndriiSherman** [comment]
Fixed in drizzle-kit@0.20.7

**Comment by cosileone** [comment]
> So it should be fixed on `drizzle-kit@beta` now.
> 
> @dane-stevens for your specific case, you will still see a few migrations generated by drizzle-kit on every push. This one is related to your use of
> 
> ```ts
> .default(sql`CURRENT_TIMESTAMP(3) ON UPDATE CURRENT_TIMESTAMP(3)`)
> ```
> 
> I would not recommend using `ON UPDATE` in default unless you really need it and are fine with having those changes triggered on every push. I'll review this [PR](https://github.com/drizzle-team/drizzle-orm/pull/1114), add a custom onUpdate SQL, and make the necessary changes to drizzle-kit this week. This way, you can have native support for the custom onUpdate function.

By the way, how do you recommend setting dates ON UPDATE if not this way?
